vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -44,6 +56,16 @@ def _filter_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
     return series.isin(value)
 
 
+def _filter_between_date(series: pd.Series, value: List[float]) -> pd.Series:
+    date_series, date_value = [_convert_to_date(value=elem) for elem in (series, value)]
+    return _filter_between(series=date_series, value=date_value)
+
+
+def _filter_isin_date(series: pd.Series, value: MultiValueType) -> pd.Series:
+    date_series, date_value = [_convert_to_date(value=elem) for elem in (series, value)]
+    return _filter_isin(series=date_series, value=date_value)
+
+
```

Comment:
I would like to get an opinion on the two added filter functions. Does it seem like the right step or not? 

I decided to use two new filter functions instead of checking if `date_series` and `date_value` can be converted to datetime format (again) (e.g. inside the `_filter_between` and `_filter_isin` functions or inside the `_actions_utils.py`), because it doesn't seem like the best practice checking the same thing multiple times. Currently, this check is only conducted once, as a part of the `Filter.pre_build` stage (`_set_column_type` method).

Also, checking if the filter_function input argument `series` can be converted to the datetime could cause bugs since the input `series` is just the part of the already filtered data_frame (if more filters are on the screen).

Reply:
Both approaches seem correct to me. The only question is, do we want to treat any type of date format as a date column or only treat the "YYYY-MM-DD" format as a date column, otherwise it's just a string and a categorical filter?

@antonymilne  @huong-li-nguyen @nadijagraca
What is you opinion? Currently, the date_picker will automatically be set as a Filter selector if the filtered column contains any date format (data such as: ["1-1-1952", "2-1-1952", "3-1-1952"])

Reply:
If we already have a functionality to treat any type of date format as a date column I would leave it, but Max's approach works for me as well.

Reply:
Here is a brief description of the conclusion:
 - If the column is `pd.api.types.is_datetime64_any_dtype` then it should go to `DatePicker` by default; otherwise it is treated as categorical so will go to `Dropdown` by default
 - No need to look for strings in `yyyy-mm-dd` or any other format. Just treat them as categorical.